keyboard_rewrite/src/lib/y2_keyboard/keyboard_layout.rb

```diff
@@ -0,0 +1,26 @@
+require "cheetah"
+
+module Y2Keyboard
+  class KeyboardLayout
+    attr_reader :code
+    attr_reader :description
+
+    LAYOUT_CODE_DESCRIPTIONS = {
+      "es" => "Spanish",
+      "fr" => "French",
+      "us" => "English (US)"
+    }
+
+    def initialize(code, description)
+      @code = code
+      @description = description
+    end
+
+    def self.load
+      raw_layouts = Cheetah.run("localectl", "list-keymaps", stdout: :capture)
+      layout_codes = raw_layouts.lines.map { |string| string.split.first }
```

Comment:
I would use a shorter name (`codes` should be enough). About the `string` argument for the block, the name tell me nothing. You could use `code` (I tend to use a single letter).

On the other hand, if you only want to remove the "\n", you should better use `strip`.

```ruby
codes = raw_layouts.lines.map { |c| c.strip }
```

And you can turn this into:

```ruby
codes = raw_layouts.map(&:strip)
```